### PR TITLE
[terra-paginator] Update Terra Doc Site

### PR DESCRIPTION
**Pull request by chenyixuan625**
### Summary

**What was changed:**
Added an accessibility guide for the Paginator component

**Why it was changed:**
So users have a reference for accessibility guidelines and changes we've made to the paginator component.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8576 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by chenyixuan625** [comment]
@scottwilmarth UX Accessibility icons are added to the ARIA props:
Screenshot 2023-09-20 at 8 45 59 AM
Examples and the accessibility doc are updated as we discussed as well. Good to be reviewed again.

File: packages/terra-paginator/src/ControlledProgressivePaginator.jsx [hunk 1]
```diff
@@ -44,7 +46,8 @@ const propTypes = {
    */
   intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
   /**
-   * Allows user to set custom page label. _(usage note: User must pass translated text)_.
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue)
+   * Allows user to set custom page label. _(usage note: User must pass translated text)_. It should not contain page number and total count details, which will lead to conflicts with build-in accessibility ARIA contexts.
```
Comment (chenyixuan625) [inline packages/terra-paginator/src/ControlledProgressivePaginator.jsx:50]: @scottwilmarth I updated the description of the _pageLabel_ prop based on our discussion. Please let me know if it looks good to you

File: packages/terra-paginator/src/ProgressivePaginator.jsx [hunk 2]
```diff
@@ -44,7 +46,8 @@ const propTypes = {
    */
   intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
   /**
-   * Allows user to set custom page label. _(usage note: User must pass translated text)_.
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue)
+   * Allows user to set custom page label. _(usage note: User must pass translated text)_. It should not contain page number and total count details, which will lead to conflicts with build-in accessibility ARIA contexts.
```
Comment (krycunn) [inline packages/terra-paginator/src/ProgressivePaginator.jsx:50]: same as above: "built-in"

File: packages/terra-paginator/src/ControlledProgressivePaginator.jsx [hunk 3]
```diff
@@ -44,7 +46,8 @@ const propTypes = {
...
```
[the remark this answers is not in the document]
Comment (chenyixuan625) [reply packages/terra-paginator/src/ControlledProgressivePaginator.jsx]: Fixed: https://github.com/cerner/terra-core/pull/3910/commits/105fb8eceb374e0f1a8878386dc205fc4000efb9

**Review by scottwilmarth — APPROVED** [review]

File: packages/terra-core-docs/src/terra-dev-site/doc/paginator/AccessibilityGuide.6.doc.mdx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+From the accessibility perspective, some additional context is required for users to understand what is happening and where they are when interacting with the paginator.
+
+In order to provide the additional context, a visually hidden text string is constructed and associated with the paginator container.
+This visually hidden string will be read out by the screen reader when the user navigates using the paginator links. The visually hidden string is empty so nothing is read out in the initial status.
+
+Depending on the optional `totalCount` prop set or not, the constructed visually hidden text varies:
+
+##### Code Example:
+
+Using the `totalCount` prop is best practice and should be used if at all possible. This example shows when the optional `totalCount` prop is set, the paginator would read out a string like "Page {pageNumber} selected. Page {pageNumber} of {pageNumberTotal}":
```
Comment (sdadn) [inline packages/terra-core-docs/src/terra-dev-site/doc/paginator/AccessibilityGuide.6.doc.mdx:63]:
```suggestion
Using the `totalCount` prop is best practice and should be used whenever possible. This example shows when the optional `totalCount` prop is set, the paginator would read out a string like "Page {pageNumber} selected. Page {pageNumber} of {pageNumberTotal}":
```

File: packages/terra-core-docs/src/terra-dev-site/doc/paginator/AccessibilityGuide.6.doc.mdx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+##### Code Example:
+
+This example shows how to set up the optional `ariaLabel` prop in order to provide meaningful context for the paginator:
+
+<ControlledPaginatorNoPagesExample />
+
+#### Provide additional context to assistive technology users after page nagivation
+
+From the accessibility perspective, some additional context is required for users to understand what is happening and where they are when interacting with the paginator.
```
Comment (sdadn) [inline packages/terra-core-docs/src/terra-dev-site/doc/paginator/AccessibilityGuide.6.doc.mdx:54]:
```suggestion
From an accessibility perspective, some additional context is required for users to understand what is happening and where they are when interacting with the paginator.
```

File: packages/terra-core-docs/src/terra-dev-site/doc/paginator/AccessibilityGuide.6.doc.mdx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+In order to provide the additional context, a visually hidden text string is constructed and associated with the paginator container.
+This visually hidden string will be read out by the screen reader when the user navigates using the paginator links. The visually hidden string is empty so nothing is read out in the initial status.
```
Comment (sdadn) [inline packages/terra-core-docs/src/terra-dev-site/doc/paginator/AccessibilityGuide.6.doc.mdx:57]:
```suggestion
This visually hidden string will be read out by a screen reader when the user navigates using the paginator links. The visually hidden string is empty so nothing is read out in the initial status.
```

**Review by sdadn — COMMENTED** [review]
Some minor phrasing and grammar updates.

**Review by sycombs — APPROVED** [review]

**Pull request merged**